libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -570,7 +571,7 @@ - (void)beginUserAgentFlow {
                     NSDictionary *dict = (NSDictionary *)json;
                     if (dict[kSFOAuthAccessToken]) {
                         NSString *escapedString = [approvalUrl stringByURLEncoding];
-                        NSString* approvalUrl = [NSString stringWithFormat:@"%@://%@/secur/frontdoor.jsp?sid=%@&retURL=%@", self.credentials.protocol, self.credentials.instanceUrl, dict[kSFOAuthAccessToken],escapedString];
+                        NSString* approvalUrl = [NSString stringWithFormat:@"%@://%@/secur/frontdoor.jsp?sid=%@&retURL=%@", self.credentials.protocol, self.credentials.domain, dict[kSFOAuthAccessToken],escapedString];
```

Comment:
Ah yes, I got that wrong.  We should be populating the rest of the available credentials properties from the output here, including `instance_url` and `community_url`, which should drive the frontdoor base URL.  I'll update accordingly.